src/java.desktop/share/classes/javax/swing/text/TextAction.java

```diff
@@ -104,7 +100,7 @@ protected final JTextComponent getTextComponent(ActionEvent e) {
      * @return the augmented list
      */
     public static final Action[] augmentList(Action[] list1, Action[] list2) {
-        Hashtable<String, Action> h = new Hashtable<String, Action>();
+        HashMap<String, Action> h = new HashMap<String, Action>();
```

Comment:
I wonder will this can introduce some behavior change related to the null values in the passed arrays? I assume before the fix we throw an exception and after we will accept null?

Reply:
You are right.
If one of `binding.actionName` is `null` - old code throw NullPointerException (from HashTable.get).
In new code such binding will be silently ignored.
As I understand we have 3 options now:
1. revert changes in this file
2. consider risk as low and proceed with current changes
3. insert Object.requireNonNull to _simulate_ old behavior

I think requoreNonNull is the best. What do you think?

Reply:
It depends on how the fix will look like, probably we can leave it as is.